api/src/worker/uploads.js

```diff
@@ -139,7 +166,154 @@ const normalizeDivision = (shitShowDivisionNameCanBeAnythingWTFPS) => {
+    match_shooters.map((s) => [s.sh_id, [s.sh_fn, s.sh_ln].filter(Boolean).join(" ")])
+  );
+  const classifiersMap = Object.fromEntries(
+    match_stages
+      .filter((s) => !!s.stage_classifiercode)
+      .map((s) => [s.stage_uuid, s.stage_classifiercode])
+  );
+
+  const { match_scores } = scoresJson;
+  const stageScoresMap = match_scores.reduce((acc, cur) => {
+    const curStage = acc[cur.stage_uuid] || {};
+    cur.stage_stagescores.forEach((cs) => {
+      curStage[cs.shtr] = cs;
+    });
+    acc[cur.stage_uuid] = curStage;
+    return acc;
+  }, {});
+
+  const scores = match_scores
+    .filter((ms) => {
+      const { stage_uuid } = ms;
+      const classifierCode = classifiersMap[stage_uuid];
+      return classifiersMap[stage_uuid] !== undefined && classifierCode.match(/SC-10[0-8]/g);
+    })
+    .map((ms) => {
+      const { stage_uuid, stage_stagescores } = ms;
+      const classifier = classifiersMap[stage_uuid];
+
+      return stage_stagescores.filter((ss) => {
+        const divisionCode = shootersDivisionMap[ss.shtr]?.toUpperCase();
+        return divisionCode !== undefined && Object.keys(PeakTimesMap).find((div) => div === divisionCode) !== undefined;
+      }).map((ss) => {
+        // str is the array of all strings for the stage
+        // e.g. [7, 5.46, 6.17, 23.13]
+        const strings = ss.str;
+
+        // penss is a two-dimensional array of the COUNT of all penalties, by index, on the stage.
+        // e.g.
+        //[
+        //  [
+        //    1,
+        //    0,
+        //    0,
+        //    0
+        //  ]
+        const pens = ss.penss;
+        const penaltyCount = pens.flat().reduce((p, c) => p + c , 0)
+        const detailedScores = stageScoresMap[stage_uuid]?.[ss.shtr] || {};
+
+        const adjustedStrings = strings.map((s, idx) => {
+          const penCountsForString = pens[idx];
+          // Multiply the count of each penalties by their value, and sum the result.
+          const totalStringPenalties = penCountsForString.reduce((p, c, idx) => p + (c * match_penalties[idx].pen_val), 0);
+          const adjustedStringTotal = s + totalStringPenalties;
+          // Strings max out at 30 seconds in SCSA.
+          return Math.min(30, adjustedStringTotal);
+        })
+          .sort(((x, y) => y - x));
+
+        const memberNumber = shootersMap[ss.shtr]?.toUpperCase();
+        const divisionCode = shootersDivisionMap[ss.shtr]?.toUpperCase();
+        const division = scsaDivisionWithPrefix(divisionCode);
+
+        const modifiedDate = new Date(detailedScores.mod);
+        const modified = Number.isNaN(modifiedDate.getTime()) ? undefined : modifiedDate;
+
+        // Worst string (front of the array) dropped.
+        const bestNStrings = adjustedStrings.slice(1);
+
+        const stageTotal = Number(bestNStrings.reduce((p, c) => p + c, 0).toFixed(2));
+
+        const classifierPeakTime = scsaPeakTime(divisionCode, classifier);
+        return {
+          hf: stageTotal,
```

Comment:
No longer populating `hf`/`hhf` here.  Instead, supplying a new field `stagePeakTimeSecs`.  The existing `stageTimeSecs` field should work for the denominator since the schema is denormalized anyway.